mlir/include/mlir/Dialect/MIOpen/MIOpenOps.td

```diff
@@ -150,28 +177,38 @@ def MIOpen_SubviewOp:
     The memref must be on either LDS (address space 3) or private (address
     space 5).
   }];
+  let assemblyFormat = [{
+    `(` $input `,` $offset `)` attr-dict `:` type($input) `to` type($output)
+  }];
 }
 
 // TBD: eventually replace this with linalg.fill?
 def MIOpen_FillOp:
-    MIOpen_Op<"fill">,
+    MIOpen_Op<"fill", [AllElementTypesMatch<["input", "value"]>]>,
     Arguments<(ins AnyMemRef:$input,
                    AnyTypeOf<[AnyInteger, AnyFloat]>:$value)> {
   let summary = "Fill memory with constant value on GPU";
   let description = [{
     The `miopen.fill` op fills a memref on GPU with a constant value.
   }];
+  let assemblyFormat = [{
+    `(` operands `)` attr-dict `:` type(operands)
```

Comment:
from the updated unit test, the syntax feels redundant to me.
for a memref filled with `f32` we should expect the constant value be of the same type `f32`.

Reply:
Agreed, though the declaritive syntax doesn't currently have a mechanism for imposing that constraint.

We could go back to a custom parser/printer for this one, or live with the redudancy for now and send in a patch to support this type of reasoning.